apps/src/templates/instructions.jsx

```diff
@@ -0,0 +1,59 @@
+var msg = require('../locale');
+
+module.exports = React.createClass({
+
+  propTypes: {
+    puzzleTitle: React.PropTypes.string,
+    instructions: React.PropTypes.string,
+    instructions2: React.PropTypes.string,
+    renderedMarkdown: React.PropTypes.string,
+    authoredHints: React.PropTypes.array,
+    markdownClassicMargins: React.PropTypes.bool,
+    aniGifURL: React.PropTypes.string
+  },
+
+  render: function () {
+
+    // The logic here needs some serious cleanup
```

Comment:
Maybe provide a little detail on how this might be cleaner, so that future you (or someone else) isn't just scared of this code without knowing how to become less scared :-P